Content.Server/GameObjects/Components/GUI/StrippableComponent.cs

```diff
@@ -0,0 +1,371 @@
+                    return false;
+
+                if (item == null)
+                {
+                    _notifyManager.PopupMessageCursor(user, Loc.GetString("You aren't holding anything!"));
+                    return false;
+                }
+
+                if (!userHands.CanDrop(userHands.ActiveHand!))
+                {
+                    _notifyManager.PopupMessageCursor(user, Loc.GetString("You can't drop that!"));
+                    return false;
+                }
+
+                if (!inventory.HasSlot(slot))
+                    return false;
+
+                if (inventory.TryGetSlotItem(slot, out ItemComponent _))
+                {
+                    _notifyManager.PopupMessageCursor(user, Loc.GetString("{0:They} already have something there!", Owner));
+                    return false;
+                }
+
+                if (!inventory.CanEquip(slot, item))
+                {
+                    _notifyManager.PopupMessageCursor(user, Loc.GetString("{0:They} cannot equip that there!", Owner));
+                    return false;
+                }
+
+                return true;
+            }
+
+            var doAfterSystem = EntitySystem.Get<DoAfterSystem>();
+
+            var doAfterArgs = new DoAfterEventArgs(user, StripDelay, CancellationToken.None, Owner)
+            {
+                ExtraCheck = Check,
+                BreakOnStun = true,
+                BreakOnDamage = true,
+                BreakOnTargetMove = true,
+                BreakOnUserMove = true,
+                NeedHand = true,
+            };
+
+            var result = await doAfterSystem.DoAfter(doAfterArgs);
+            if (result != DoAfterStatus.Finished) return;
+
+            userHands.Drop(item!.Owner, false);
+            inventory.Equip(slot, item!.Owner);
+
+            UpdateSubscribed();
+        }
+
+        /// <summary>
+        ///     Places item in user's active hand in one of the entity's hands.
+        /// </summary>
+        private async void PlaceActiveHandItemInHands(IEntity user, string hand)
+        {
+            var hands = Owner.GetComponent<HandsComponent>();
+            var userHands = user.GetComponent<HandsComponent>();
+            var item = userHands.GetActiveHand;
+
+            bool Check()
+            {
+                if (!ActionBlockerSystem.CanInteract(user))
+                    return false;
+
+                if (item == null)
+                {
+                    _notifyManager.PopupMessageCursor(user, Loc.GetString("You aren't holding anything!"));
+                    return false;
+                }
+
+                if (!userHands.CanDrop(userHands.ActiveHand!))
+                {
+                    _notifyManager.PopupMessageCursor(user, Loc.GetString("You can't drop that!"));
+                    return false;
+                }
+
+                if (!hands.HasHand(hand))
+                    return false;
+
+                if (hands.TryGetItem(hand, out var _))
```

Comment:
You don't need the var keyword here, though it would admittedly be better to have a `HasItem` method.

```suggestion
                if (hands.TryGetItem(hand, out _))
```